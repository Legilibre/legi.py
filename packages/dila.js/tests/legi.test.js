//const { toMatchDiffSnapshot } = require("snapshot-diff");

const Legi = require("../src");
const markdown = require("../src/markdown");
const html = require("../src/html");

const legi = new Legi("./legi.sqlite");

afterAll(() => {
  legi.close();
});

// medailles LEGITEXT000006070666
// travail LEGITEXT000006072050
// PI LEGITEXT000006069414
const CODE_TEST = "LEGITEXT000006070666";

jest.setTimeout(20000);

it("getSection", async () => {
  expect.assertions(1);
  const res = await legi.getSection({ parent: "LEGISCTA000006177833", date: "2018-05-03" });
  expect(res).toMatchSnapshot();
});

it("getSection LEGISCTA000006198560-2017-01-01 should return a single L2232-13", async () => {
  expect.assertions(2);
  const res = await legi.getSection({ parent: "LEGISCTA000006198560", date: "2017-01-01" });
  const count = res.children.filter(node => node.data.num === "L2232-13").length;
  expect(count).toBe(1);
  expect(JSON.stringify(res)).toMatchSnapshot();
});

//cid="LEGITEXT000006072050" and date_fin like '2016-%'
//LEGIARTI000030730068  LEGISCTA000030730058  R2151-1 MODIFIE 2015-06-14  2016-10-23  AUTONOME
//R2151-1
//2999-01-01

it("getSection: article en vigueur doit apparaitre", async () => {
  expect.assertions(2);
  const res = await legi.getSection({ id: "LEGISCTA000030730058", date: "2016-01-01" });
  const count = res.children.filter(node => node.data.num === "R2151-1").length;
  expect(count).toBe(1);
  expect(res).toMatchSnapshot();
});

it("getSection: article pas encore en vigueur ne doit pas apparaitre", async () => {
  expect.assertions(2);
  const res = await legi.getSection({ id: "LEGISCTA000030730058", date: "2015-01-01" });
  const count = res.children.filter(node => node.data.num === "R2151-1").length;
  expect(count).toBe(0);
  expect(res).toMatchSnapshot();
});

it("getSection: article pas encore abrogé doit apparaitre", async () => {
  expect.assertions(2);
  const res = await legi.getSection({ id: "LEGISCTA000029978970", date: "2015-02-01" });
  const count = res.children.filter(node => node.data.num === "R958-32").length;
  expect(count).toBe(1);
  expect(res).toMatchSnapshot();
});

it("getSection: article abrogé ne doit pas apparaitre", async () => {
  expect.assertions(2);
  const res = await legi.getSection({ id: "LEGISCTA000029978970", date: "2017-01-01" });
  const count = res.children.filter(node => node.data.num === "R958-32").length;
  expect(count).toBe(0);
  expect(res).toMatchSnapshot();
});

it("getCodesList", async () => {
  expect.assertions(4);
  const res = await legi.getCodesList();
  expect(typeof res.length).not.toBe("undefined");
  expect(typeof res).toBe("object");
  expect(res.length).toBeGreaterThan(10);
  expect(res).toMatchSnapshot();
});

it("getCodeVersions", async () => {
  expect.assertions(4);
  const res = await legi.getCodeVersions({ cid: CODE_TEST });
  expect(typeof res.length).not.toBe("undefined");
  expect(typeof res).toBe("object");
  expect(res.length).toBeGreaterThan(10);
  expect(res).toMatchSnapshot();
});

it(`getCode ${CODE_TEST}-2018-04-01`, async () => {
  expect.assertions(1);
  const res = await legi.getCode({ id: CODE_TEST, date: "2018-04-01" });
  expect(res).toMatchSnapshot();
});

it(`getCode ${CODE_TEST}-1970-01-01`, async () => {
  expect.assertions(1);
  const res = await legi.getCode({ id: CODE_TEST, date: "1970-01-01" });
  expect(res).toMatchSnapshot();
});

it("getJORF JORFTEXT000036386252", async () => {
  expect.assertions(1);
  const res = await legi.getJORF("JORFTEXT000036386252");
  expect(res).toMatchSnapshot();
});

const codeTest = legi.getCode(CODE_TEST);

it("getCode then markdown", async () => {
  expect.assertions(1);
  const res = await codeTest.then(markdown).catch(console.log);
  expect(res).toMatchSnapshot();
});

it("getCode then html", async () => {
  expect.assertions(1);
  const res = await codeTest.then(html).catch(console.log);
  expect(res).toMatchSnapshot();
});
