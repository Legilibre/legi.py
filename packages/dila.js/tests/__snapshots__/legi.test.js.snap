// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSection: article abrogé ne doit pas apparaitre 1`] = `
Object {
  "children": Array [],
  "data": Object {
    "cid": "LEGITEXT000006071367",
    "commentaire": null,
    "dossier": "code_en_vigueur",
    "id": "LEGISCTA000029978970",
    "mtime": 1514576002,
    "parent": "LEGISCTA000032717981",
    "titre_ta": "Section 2 : Dispositions particulières à l'île de Clipperton",
  },
  "type": "section",
}
`;

exports[`getSection: article en vigueur doit apparaitre 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "Pour l'application du 4° de l'article L. 2151-1, une organisation professionnelle d'employeurs issue de la fusion d'organisations professionnelles d'employeurs préexistantes peut se prévaloir de l'ancienneté acquise antérieurement à la fusion par la plus ancienne de ces dernières dans le champ professionnel et géographique correspondant au niveau pour lequel la représentativité est demandée.",
        "cid": "LEGITEXT000006072050",
        "date_debut": 2015-06-13T22:00:00.000Z,
        "date_fin": 2016-10-22T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "MODIFIE",
        "id": "LEGIARTI000030730068",
        "mtime": 1477335569,
        "nota": "",
        "num": "R2151-1",
        "parent": "LEGISCTA000030730058",
        "position": 0,
        "section": "LEGISCTA000030730058",
        "titre_ta": "Article R2151-1",
        "type": "AUTONOME",
      },
      "type": "article",
    },
  ],
  "data": Object {
    "cid": "LEGITEXT000006072050",
    "commentaire": null,
    "dossier": "code_en_vigueur",
    "id": "LEGISCTA000030730058",
    "mtime": 1477335586,
    "parent": "LEGISCTA000030730052",
    "titre_ta": "Chapitre Ier : Critères de représentativité &#13;
&#13;
&#13;
&#13;
&#13;
",
  },
  "type": "section",
}
`;

exports[`getSection: article pas encore abrogé doit apparaitre 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>Sans préjudice des dispositions d'accords spécifiques conclus avec des Etats tiers, l'exercice de la pêche par des navires autorisés battant pavillon d'un Etat étranger dans la zone économique de l'île de Clipperton est subordonné à l'octroi d'une autorisation dans les conditions prévues à la présente section. <br/>Les navires d'assistance étrangers opérant dans la zone économique exclusive de Clipperton doivent également disposer d'une autorisation et sont soumis aux mêmes obligations. <br/>L'exercice de la pêche par d'autres navires étrangers que ceux disposant d'une autorisation est interdite. <br/>Les dispositions de la présente section ne sont pas applicables à l'exercice de la pêche scientifique. <br/>Le silence gardé par l'autorité administrative mentionnée à l'article R. * 911-3 pendant un délai de deux mois sur les demandes d'autorisation mentionnées au présent article vaut décision de rejet.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "MODIFIE",
        "id": "LEGIARTI000029978778",
        "mtime": 1467316411,
        "nota": "",
        "num": "R958-26",
        "parent": "LEGISCTA000029978970",
        "position": 4,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-26",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>Les autorisations sont délivrées par décision de l'autorité mentionnée à l'article R. * 911-3, après avis du ministre des affaires étrangères, du ministre chargé des pêches maritimes et de l'aquaculture marine et du ministre chargé de l'outre-mer. <br/>Le cas échéant, cette autorité fixe par arrêté, après avis conforme de ces ministres, le nombre maximal d'autorisations susceptibles d'être délivrées, en tenant compte notamment des capacités biologiques de la zone concernée. <br/>Il peut également fixer par arrêté une limite de capture par navire.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "ABROGE",
        "id": "LEGIARTI000029978780",
        "mtime": 1467226434,
        "nota": "",
        "num": "D958-27",
        "parent": "LEGISCTA000029978970",
        "position": 5,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article D958-27",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>Les informations qui doivent être communiquées par l'armement lors de la demande d'autorisation ainsi que la procédure de délivrance sont définies par arrêté de l'autorité mentionnée à l'article R. * 911-3, après avis conforme du ministre des affaires étrangères, du ministre chargé des pêches maritimes et de l'aquaculture marine et du ministre chargé de l'outre-mer. <br/>Les autorisations de pêche sont délivrées après vérification de la capacité juridique, économique, financière et technique de l'armement bénéficiaire en tenant compte notamment : <br/>1° D'un lien économique réel du navire avec le territoire de l'Etat dont il bat le pavillon, notamment de la direction et du contrôle des navires à partir d'un établissement stable situé sur le territoire de l'Etat dont le navire bat le pavillon ; <br/>2° Des antériorités dans la pêcherie ; <br/>3° Des orientations du marché ; <br/>4° Des équilibres socio-économiques ; <br/>5° De l'engagement d'embarquer un contrôleur de pêche ou un observateur scientifique, si le préfet en fait la demande ; <br/>6° De la participation de l'armateur à des campagnes expérimentales visant à atténuer l'impact des activités de pêche sur l'environnement ou à des initiatives tendant à la protection de la ressource et de l'environnement ; <br/>7° Des infractions éventuellement commises lors des années précédentes ; <br/>8° De l'éventuelle inscription du navire ou de l'armateur sur une liste de navires ou d'opérateurs impliqués dans des activités de pêche illégale (pêche illicite, non déclarée et non réglementée) établie par l'Union européenne ou par une organisation régionale de gestion des pêches.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "ABROGE",
        "id": "LEGIARTI000029978782",
        "mtime": 1467226434,
        "nota": "",
        "num": "R958-28",
        "parent": "LEGISCTA000029978970",
        "position": 6,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-28",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>Chaque autorisation peut déterminer, en particulier, la période et les zones de pêche autorisées, les espèces ou groupes d'espèces concernés et les engins de pêche autorisés.<br/>Un navire détenteur d'une autorisation doit, pour pouvoir pêcher, se conformer à la réglementation applicable, aux mesures de conservation ou de gestion applicables adoptées par une organisation régionale de gestion des pêches compétente et aux prescriptions techniques annexées à l'autorisation.<br/>Ces prescriptions techniques peuvent fixer notamment les modalités :<br/>1° D'interdiction de pêche de certaines espèces ou d'utilisation de certains engins de pêche ;<br/>2° De déclaration des captures des espèces principalement ciblées et des prises accessoires ;<br/>3° D'atténuation des captures accidentelles d'oiseaux, de reptiles et de mammifères marins ;<br/>4° D'interdiction de certaines espèces ;<br/>5° D'interdiction de rejet en mer d'objets en matière non dégradable ;<br/>6° De déclaration d'entrée et de sortie de la zone économique de Clipperton ;<br/>7° D'identification et de suivi par tout moyen des navires autorisés ;<br/>8° D'embarquement ou de transfert et d'accueil d'un contrôleur de pêche ou d'un observateur scientifique.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "MODIFIE",
        "id": "LEGIARTI000029978784",
        "mtime": 1467316410,
        "nota": "",
        "num": "R958-29",
        "parent": "LEGISCTA000029978970",
        "position": 7,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-29",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>La durée de validité de l'autorisation de pêche, qui ne peut excéder une année, cesse au 31 décembre de l'année de délivrance.<br/>L'autorisation doit être conservée à bord en permanence ; elle ne peut être ni cédée ni vendue.<br/>Le refus opposé à une demande d'autorisation doit être motivé et notifié au demandeur.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "MODIFIE",
        "id": "LEGIARTI000029978786",
        "mtime": 1467316410,
        "nota": "",
        "num": "R958-30",
        "parent": "LEGISCTA000029978970",
        "position": 8,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-30",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>La délivrance d'une autorisation peut donner lieu au versement d'une contrepartie financière par l'armement, selon des modalités fixées par arrêté de l'autorité mentionnée à l'article R. * 911-3, après avis conforme du ministre des affaires étrangères, du ministre chargé des pêches maritimes et de l'aquaculture marine et du ministre chargé de l'outre-mer. <br/>La contrepartie financière annuelle comprend une part fixe, qui doit être réglée avant la délivrance des autorisations, et une part variable. <br/>La part fixe comprend un montant couvrant un poids forfaitaire de captures dans la zone économique de Clipperton. Elle peut également comprendre une redevance pour frais de surveillance et d'observation. <br/>La part variable s'applique pour les captures au-delà d'un certain tonnage et doit être payée dans un délai de deux mois après la fin de la campagne et au plus tard le 1er novembre de l'année en cours. <br/>Les captures accessoires, dont la comptabilité doit être tenue, font également l'objet d'une contrepartie financière. <br/>Tout manquement des armements aux obligations du présent article entraîne la suspension immédiate de la licence par l'autorité qui l'a délivrée.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "ABROGE",
        "id": "LEGIARTI000029978788",
        "mtime": 1467226433,
        "nota": "",
        "num": "D958-31",
        "parent": "LEGISCTA000029978970",
        "position": 9,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article D958-31",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>L'autorisation peut être retirée sans indemnité par l'autorité mentionnée à l'article R. * 911-3 après que l'armement concerné a été mis en mesure de présenter ses observations, dans les cas où : <br/>1° Les dispositions de la présente section ou prises en application de la présente section ou les prescriptions techniques annexées à l'autorisation n'ont pas été respectées ; <br/>2° Le contrôleur de pêche ou l'observateur scientifique embarqué a été entravé de quelque manière que ce soit dans l'exercice de ses fonctions ou missions, y compris du fait d'un manque de coopération de la part de l'équipage ; <br/>3° Les caractéristiques ou le mode d'exploitation du navire ont été modifiés et ne répondent plus aux conditions fixées pour la délivrance de l'autorisation ; <br/>4° Le navire a été vendu ou cédé à un titre quelconque.</p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "ABROGE",
        "id": "LEGIARTI000029978790",
        "mtime": 1467226434,
        "nota": "",
        "num": "R958-32",
        "parent": "LEGISCTA000029978970",
        "position": 10,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-32",
        "type": "AUTONOME",
      },
      "type": "article",
    },
    Object {
      "children": undefined,
      "data": Object {
        "bloc_textuel": "<p><br/>A défaut de représentants des administrations prévues à l'article 31 du décret n° 71-360 du 6 mai 1971 portant application de la loi n° 68-1181 du 30 décembre 1968 relative à l'exploration du plateau continental et à l'exploitation de ses ressources naturelles pour siéger à la commission d'études des programmes mentionnée à l'article 8 de ce décret, des représentants des administrations concernées ou des organismes scientifiques compétentes pour le territoire leur sont substitués par l'autorité désignée à l'article R. * 911-3. </p>",
        "cid": "LEGITEXT000006071367",
        "date_debut": 2014-12-31T23:00:00.000Z,
        "date_fin": 2016-06-30T22:00:00.000Z,
        "dossier": "code_en_vigueur",
        "etat": "MODIFIE",
        "id": "LEGIARTI000029978792",
        "mtime": 1467316410,
        "nota": "",
        "num": "R958-33",
        "parent": "LEGISCTA000029978970",
        "position": 11,
        "section": "LEGISCTA000032717973",
        "titre_ta": "Article R958-33",
        "type": "AUTONOME",
      },
      "type": "article",
    },
  ],
  "data": Object {
    "cid": "LEGITEXT000006071367",
    "commentaire": null,
    "dossier": "code_en_vigueur",
    "id": "LEGISCTA000029978970",
    "mtime": 1514576002,
    "parent": "LEGISCTA000032717981",
    "titre_ta": "Section 2 : Dispositions particulières à l'île de Clipperton",
  },
  "type": "section",
}
`;

exports[`getSection: article pas encore en vigueur ne doit pas apparaitre 1`] = `
Object {
  "children": Array [],
  "data": Object {
    "cid": "LEGITEXT000006072050",
    "commentaire": null,
    "dossier": "code_en_vigueur",
    "id": "LEGISCTA000030730058",
    "mtime": 1477335586,
    "parent": "LEGISCTA000030730052",
    "titre_ta": "Chapitre Ier : Critères de représentativité &#13;
&#13;
&#13;
&#13;
&#13;
",
  },
  "type": "section",
}
`;
